src/interfaces/init.h

```diff
@@ -0,0 +1,105 @@
+//! process.
+//!
+//! The interfaces::Init interface is similar to other cross-process C++
+//! interfaces like interfaces::Node, interfaces::Wallet, interfaces::Chain and
+//! interfaces::ChainClient, providing virtual methods that can be called from
+//! other processes. What makes it special is that unlike other interfaces which
+//! are not implemented by every process—interfaces::Node is only implemented by
+//! the node process and interfaces::Wallet is only implemented by the wallet
+//! process—interfaces::Init is implemented by every process that supports being
+//! spawned, and it is the initial interface returned by the
+//! IpcProtocol::connect(fd) method, allowing the parent process to control the
+//! child process after the connection is established. The interfaces::Init
+//! interface has methods that allow the parent process to get access to every
+//! interface supported by the child process, and when the parent process frees
+//! the interfaces::Init object, the child process shuts down.
+//!
+//! When spawning a new process, the steps are:
+//!
+//! 1. Client process calls IpcProcess::spawn(), which spawns a new process and
+//!    returns a socketpair file descriptor for communicating with it.
+//! 2. Client process calls IpcProtocol::connect() passing the socketpair
+//!    descriptor, which returns a client Init proxy object calling remote Init
+//!    interface methods.
+//! 3. Client process calls client Init proxy object methods to make new client
+//!    proxy objects calling other remote interfaces. It can also destroy the
+//!    Init object to shut down the spawned process.
+//! 4. Spawned process calls IpcProcess::serve(), to read command line arguments
+//!    and determine that it is a spawned process and what socketpair file
+//!    descriptor it should use. The spawned process then calls
+//!    IpcProtocol::serve() to handle incoming requests from the socketpair and
+//!    invoke Init interface methods and eventually shut down and destroy the
+//!    Init interface when the connection is closed.
+//!
+//! When connecting to an existing process, the steps are similar to spawning a
+//! new process, except a socket is created instead of a socketpair, and
+//! destroying an Init interface doesn't end the process, since there can be
+//! multiple connections.
+class Init
+{
+public:
+    virtual ~Init() = default;
+    virtual std::unique_ptr<Echo> makeEcho() = 0;
+};
+
+//! Specialization of the Init interface for the local process. Container for
+//! IpcProcess and IpcProtocol objects and current process information.
+class LocalInit : public Init
+{
+public:
+    LocalInit(const char* exe_name, const char* log_suffix);
+    ~LocalInit() override;
+    std::unique_ptr<Echo> makeEcho() override;
+    //! Make echo implementation for `echoipc` test RPC. Spawn new process if
+    //! supported.
+    virtual std::unique_ptr<Echo> makeEchoIpc();
```

Comment:
I think you could add a temporary comment hinting this interface as the future location of `makeNode`, `makeWalletClient`, etc, it's more helpful for reviewers to understand what it would look like after #10102.